player/lua/select.lua

```diff
@@ -61,7 +61,7 @@ local function format_track(track)
             (track["demux-w"] and track["demux-w"] .. "x" .. track["demux-h"]
              .. " " or "") ..
             (track["demux-fps"] and not track.image
-             and string.format("%.3f", track["demux-fps"]) .. " fps " or "") ..
+             and string.format("%.5g", track["demux-fps"]) .. " fps " or "") ..
```

Comment:
Use `mp.get_property_osd("current-tracks/video/demux-fps")` to get existing formatting code to take care of it.